root/collection/CollectionIndex.js

```diff
@@ -172,8 +172,18 @@ React.Element<typeof CollectionLayout> => {
     (user.id === collection.editor.id ||
       collection.collaborators.some(x => x.id === user.id));
 
+  const isRecordingCollection = collectionEntityType === 'recording';
+  const recordingMbids = isRecordingCollection && entities.length > 0
+    // $FlowIssue (both incompatible-type and prop-missing, sigh)
+    ? entities.map((entity: RecordingT) => entity.gid)
```

Comment:
Nitpick: the incompatible-type isn't really a Flow issue, since it has no way of knowing these are recordings.  I'm not sure why it's giving prop-missing there, that does seem like a Flow issue.  But I was able to work around both by casting the list to `$ReadOnlyArray<CoreEntityT>` first:
```suggestion
    ? (entities: $ReadOnlyArray<CoreEntityT>).map(entity => entity.gid)
```

Reply:
I don’t mind either way for now because it’s just a global issue in MBS JS code.

Just as a side-note: `CoreEntityT` is overused in general. It is better than using `any` but still. This is one of the main issues I encountered while working on https://github.com/metabrainz/musicbrainz-server/pull/2732 the 4 first commits of which are just about replacing a few unneeded `CoreEntityTypeT` occurrences. There are 300+ occurrences in the JS code, there are good chances that many of these are not fine-grained enough, later on generating the kind of Flow issue that you have here.

Reply:
But it already knows that `entities: $ReadOnlyArray<T>` and `T: CoreEntityT` - as such, it should know these have gids. In any case, casting to `CoreEntityT` again seems like a hack. What would be ideal is for Flow to understand that if we know `collectionEntityType` is `'recording'` then we also know that `entities` is `$ReadOnlyArray<RecordingT>`, but I'm guessing with the way we're doing this now, we can't just use the https://flow.org/en/docs/types/unions/#toc-disjoint-object-unions thing since hardcoding the `collectionEntityType` directly for all, but typing it as `T['entityType']`. I guess this might work better if the type was a proper union instead?

Reply:
Actually yeah, I missed that `props` is a disjoint object union, so we could use that:

```js
  const recordingMbids = props.collectionEntityType === 'recording' && props.entities.length > 0
    ? props.entities.map(entity => entity.gid)
    : null;
```

Reply:
I had to change the type declaration a bit, but now it's happy (we can squash on merge as needed).